packages/ui-components/src/components/Chart/PieChart.js

```diff
@@ -26,136 +26,70 @@
-  } = shapeProps;
-
-  const RADIAN = Math.PI / 180;
-  const sin = Math.sin(-RADIAN * midAngle);
-  const cos = Math.cos(-RADIAN * midAngle);
-  const isLabelOnRight = cos >= 0;
-
-  // We're drawing two lines:
-  // - extension line, extending out from pie chart at angle = midAngle
-  // - text line, horizontally from extension line endpoint to start of text
-  // For the most part they can use set lengths _but_ at near-vertical angles
-  // this can make text intersect other visual elements, so we need to do a
-  // tiny bit more math.
-
-  // Check if we're pointing dangerously straight-up or straight-down
-  // and adjust lines to suit - shorten extension line & lengthen text line
-  const adjustmentAmount = 1.5; // (just derived from eyeballing what looks ok)
-  const adjustmentFactor = Math.min(1, Math.abs(cos * adjustmentAmount));
-
-  // find start of extension line
-  const extensionStartRadius = outerRadius + 10;
-  const sx = cx + extensionStartRadius * cos;
-  const sy = cy + extensionStartRadius * sin;
-
-  // find end of extension line / start of text line
-  const extensionEndRadius = extensionStartRadius + adjustmentFactor * 20;
-  const mx = cx + extensionEndRadius * cos;
-  const my = cy + extensionEndRadius * sin;
-
-  // find end of text line
-  const horizontalTooltipLineLength = 52 + (1 - adjustmentFactor) * 30;
-  const ex = mx + (isLabelOnRight ? 1 : -1) * horizontalTooltipLineLength;
-  const ey = my;
-
-  const textAnchor = isLabelOnRight ? 'start' : 'end';
-  const valueTypeForLabel = labelType || valueType;
-  const metadataForLabel = originalItem[`${name}_metadata`];
-
-  return (
-    <g>
-      <Sector
-        cx={cx}
-        cy={cy}
-        innerRadius={innerRadius}
-        outerRadius={outerRadius}
-        startAngle={startAngle}
-        endAngle={endAngle}
-        fill={fill}
-      />
-      <Sector
-        cx={cx}
-        cy={cy}
-        startAngle={startAngle}
-        endAngle={endAngle}
-        innerRadius={outerRadius + 6}
-        outerRadius={outerRadius + 10}
-        fill={fill}
-      />
-      <path d={`M${sx},${sy} L${mx},${my} L${ex},${ey}`} stroke={fill} fill="none" />
-      <circle cx={ex} cy={ey} r={2} fill={fill} stroke="none" />
-      <text x={ex + (isLabelOnRight ? 1 : -1) * 12} y={ey} textAnchor={textAnchor} fill="#BBB">
-        {name}
-      </text>
-      <text
-        x={ex + (isLabelOnRight ? 1 : -1) * 12}
-        y={ey}
-        dy={18}
-        textAnchor={textAnchor}
-        fill="#999"
-      >
-        {formatDataValueByType({ value, metadata: metadataForLabel }, valueTypeForLabel)}
-      </text>
-    </g>
-  );
+const chartColorAtIndex = (colorArray, index) => {
+  return colorArray[index % colorArray.length];
 };
 
 export const PieChart = ({ viewContent, isExporting, isEnlarged, onItemClick }) => {
   const [activeIndex, setActiveIndex] = useState(-1);
 
   // Disable tapping charts to reveal legend labels on mobile as they do not fit and are awkwardly cropped
```

Comment:
Considering we've stripped out all the other support for mobile and tupaiay stuff, as well as changing the hover behaviour, let's remove the mobile check here too

Reply:
Yeah it's a fine balance of refactoring and updating stuff while not loosing any key functionality from tupaia. I may over step the line sometimes so it's good to ask the question. I'm keen to phase out the isMobile stuff in favour of a responsive approach but I've tried to keep it contained to just the interactions on the chart and the legend here. So yes good suggestion but I'm going to leave in the other mobile related bits of these components for now.